Rewrite PostgreSQL documentation (#2389)
Scalingo/documentation: src/_posts/databases/postgresql/2000-01-01-timescaledb.md

---
title: TimescaleDB®
nav: TimescaleDB®
modified_at: 2023-12-11 12:00:00
tags: databases postgresql timescaledb timeseries extensions
index: 11
---

TimescaleDB® is an open-source extension to PostgreSQL® designed to make SQL
scalable for time series data.

This extension adds several features and functions. You can find more
information on the [official documentation](https://docs.timescale.com/api/latest).

## TimescaleDB® at Scalingo

> {% note %}
> TimescaleDB® has some specificities which you should be aware of before getting
> started.
> {% endnote %}

Reviewer: suggestion: I wouldn't have this sentence in a not bloc. I think that the information it contains does not requires that much highlight

```suggestion
TimescaleDB® has some particularity which you should be aware of before getting
started.
```

Author: Fixed in https://github.com/Scalingo/documentation/pull/2389/commits/fc868ed5a747ffe69e0914b5fa96d9e3ab6ca37f

- There are two licenses of TimescaleDB®: **Community** edition and
  **Open-Source** edition. The latter is the only one available on Scalingo,
  hence TimescaleDB®'s functions labeled "*community*" on their documentation
  are not available on the platform.

> - TimescaleDB® requires a PostgreSQL® database version 13.6.0-1 or above.

Reviewer: suggestion: emphasize that it's a Scalingo-specific limitation?
```suggestion
- On Scalingo, TimescaleDB® requires a PostgreSQL® database version 13.6.0-1 or above.
```

- Working with a time series database usually involves the need to [compress

>   data](#downsampling) in runtime and [remove the oldest data](#configuring-data-retention).

Author: Fixed in https://github.com/Scalingo/documentation/pull/2389/commits/fc868ed5a747ffe69e0914b5fa96d9e3ab6ca37f

  We provide documentation about these topics below.

- While most documentation about PostgreSQL® remains applicable for
  TimescaleDB®, there are a few elements that differ, **especially regarding
  [dumping](#dumping-and-restoring) and [restoring](#dumping-and-restoring) a
  TimescaleDB® database**.

## Enabling TimescaleDB®

To enable TimescaleDB®:
1. [Provision a new PostgreSQL® database]({% post_url databases/postgresql/2000-01-01-getting-started %})

> 2. [Enable the `timescaledb` extension]({% post_url databases/postgresql/2000-01-01-managing %}#enabling-an-extension)

Reviewer: suggestion: I would include here the SQL command customers need to use
```suggestion
2. [Enable the `timescaledb` extension]({% post_url databases/postgresql/2000-01-01-managing %}#enabling-an-extension) with `CREATE EXTENSION timescaledb`
```

## Downsampling

### Understanding Downsampling

Downsampling consists in reducing the amount of data by compressing the old

> values.

Reviewer: issue: downsampling is not compressing. Downsampling is removing some data so that we have less fine-grained data after some times. Downsampling and compressing are two different things that people can do on TimescaleDB data.
